### PR TITLE
(PLATFORM-2857) Add a link to download original file on image pages

### PR DESCRIPTION
**Pull request by Grunny**
/cc @Wikia/core-platform-team

### CODE REVIEW

File: includes/ImagePage.php [hunk 1]
```diff
@@ -370,7 +370,7 @@ protected function openShowImage() {
 			$height_orig = $this->displayImg->getHeight( $page );
 			$height = $height_orig;
 
-			$longDesc = wfMsg( 'parentheses', $this->displayImg->getLongDesc() );
+			$longDesc = wfMessage( 'parentheses', $this->displayImg->getLongDesc() )->escaped();
```
Comment (WikiaTech) [inline includes/ImagePage.php]: MAJOR Remove this unused "$longDesc" local variable. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1481)

File: includes/ImagePage.php [hunk 2]
```diff
@@ -515,22 +515,33 @@ protected function openShowImage() {
...
-					$linktext = wfEscapeWikiText( $msgbig );
+					$linktext = htmlspecialchars( $msgbig, ENT_QUOTES );
 				}
-				$medialink = "[[Media:$filename|$linktext]]";
+				$medialink = Linker::makeMediaLinkFile( $imgTitle, $this->displayImg, $linktext );
 
 				if ( !$this->displayImg->isSafeFile() ) {
-					$warning = wfMsgNoTrans( 'mediawarning' );
-					$wgOut->addWikiText( <<<EOT
+					$warning = wfMessage( 'mediawarning' )->parse();
```
Comment (WikiaTech) [inline includes/ImagePage.php]: MAJOR Remove this unused "$warning" local variable. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1481)

**Comment by WikiaTech** [comment]
SonarQube analysis reported 56 issues
* MAJOR 18 major
* MINOR 21 minor
* INFO 17 info

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. MAJOR [ImagePage.php](https://github.com/Wikia/app/blob/eb65c79f542b279e384dbf162acc4031e93d74a3/includes/ImagePage.php): File "ImagePage.php" has 1395 lines, which is greater than 1000 authorized. Split it into smaller files. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS104)
1. MAJOR [ImagePage.php#L7](https://github.com/Wikia/app/blob/eb65c79f542b279e384dbf162acc4031e93d74a3/includes/ImagePage.php#L7): Class "ImagePage" has 28 methods, which is greater than 20 authorized. Split it into smaller classes. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1448)
1. MAJOR [ImagePage.php#L43](https://github.com/Wikia/app/blob/eb65c79f542b279e384dbf162acc4031e93d74a3/includes/ImagePage.php#L43): Take the required action to fix the issue indicated by this "FIXME" comment. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1134)
1. MAJOR [ImagePage.php#L119](https://github.com/Wikia/app/blob/eb65c79f542b279e384dbf162acc4031e93d74a3/includes/ImagePage.php#L119): Take the required action to fix the issue indicated by this "FIXME" comment. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1134)
1. MAJOR [ImagePage.php#L247](https://github.com/Wikia/app/blob/eb65c79f542b279e384dbf162acc4031e93d74a3/includes/ImagePage.php#L247): Take the required action to fix the issue indicated by this "FIXME" comment. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1134)
1. MAJOR [ImagePage.php#L284](https://github.com/Wikia/app/blob/eb65c79f542b279e384dbf162acc4031e93d74a3/includes/ImagePage.php#L284): Take the required action to fix the issue indicated by this "FIXME" comment. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1134)
1. MAJOR [ImagePage.php#L308](https://github.com/Wikia/app/blob/eb65c79f542b279e384dbf162acc4031e93d74a3/includes/ImagePage.php#L308): Take the required action to fix the issue indicated by this "FIXME" comment. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1134)
1. MAJOR [ImagePage.php#L337](https://github.com/Wikia/app/blob/eb65c79f542b279e384dbf162acc4031e93d74a3/includes/ImagePage.php#L337): The Cyclomatic Complexity of this function "openShowImage" is 34 which is greater than 20 authorized. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1541)
1. MAJOR [ImagePage.php#L337](https://github.com/Wikia/app/blob/eb65c79f542b279e384dbf162acc4031e93d74a3/includes/ImagePage.php#L337): This function "openShowImage" has 243 lines, which is greater than the 150 lines authorized. Split it into smaller functions. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS138)
1. MAJOR [ImagePage.php#L357](https://github.com/Wikia/app/blob/eb65c79f542b279e384dbf162acc4031e93d74a3/includes/ImagePage.php#L357): Remove this unused "$dirmark" local variable. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1481)

**Review by harnash — APPROVED** [review]

**Comment by ludwikkazmierczak** [comment]
Tested and Accepted

**Pull request merged**